src/software/networking/proto_multicast_listener.h

```diff
@@ -4,61 +4,44 @@
 #include <boost/bind.hpp>
 #include <string>
 
-template <class ReceiveProto>
+#include "software/networking/multicast_listener.h"
+
+template <class ReceiveProtoT>
 class ProtoMulticastListener
```

Comment:
can we have `ProtoMulticastListener` extend `MulticastListener` and override `handleDataReception`?

Reply:
I believe this would violate the [Liskov substitution principle](https://en.wikipedia.org/wiki/Liskov_substitution_principle)?

Reply:
Well it doesn't violate anything because MulticastListener doesn't guarantee any interface lmao. I suppose it's fine the way it is.